File: api-tester/src/main/java/com/revenuecat/apitester/java/CacheFetchPolicyAPI.java
package com.revenuecat.apitester.java;

import com.revenuecat.purchases.CacheFetchPolicy;

@SuppressWarnings({"unused"})
final class CacheFetchPolicyAPI {
    static void check(final CacheFetchPolicy fetchPolicy) {
        switch (fetchPolicy) {
            case CACHE_ONLY:
            case FETCH_CURRENT:
            case CACHED_OR_FETCHED:
            case NOT_STALE_CACHED_OR_CURRENT:
        }
    }
}

File: api-tester/src/main/java/com/revenuecat/apitester/kotlin/CacheFetchPolicyAPI.kt
package com.revenuecat.apitester.kotlin

import com.revenuecat.purchases.CacheFetchPolicy

@Suppress("unused", "UNUSED_VARIABLE")
private class CacheFetchPolicyAPI {

    fun check(fetchPolicy: CacheFetchPolicy) {
        when (fetchPolicy) {
            CacheFetchPolicy.CACHE_ONLY,
            CacheFetchPolicy.FETCH_CURRENT,
            CacheFetchPolicy.CACHED_OR_FETCHED,
            CacheFetchPolicy.NOT_STALE_CACHED_OR_CURRENT,
            -> {}
        }.exhaustive
    }
}
